Dockerfile

```
@@ -71,11 +62,12 @@ COPY run_tests.sh run_tests.sh
 COPY paramak paramak/
 COPY examples examples/
 COPY setup.py setup.py
+COPY setup.cfg setup.cfg
+COPY pyproject.toml pyproject.toml
 COPY tests tests/
 COPY README.md README.md
 
-# using setup.py instead of pip due to https://github.com/pypa/pip/issues/5816
-RUN python setup.py install
+RUN --mount=source=.git,target=.git,type=bind pip install --no-cache-dir -e .[tests,docs]
```

I don't entirely understand what this line is doing. I'm wondering if this means users can no longer build the docker image. Does this mean users will need to have cloned the repo and be building the docker image from within the repo and be running a Linux OS with BuildKit  installed (I think buildkit is linux only). I wonder if a docker volume mount might also work without the need for build kit

---

I think another option might be to pass the version number in as a ```--build-arg``` if that helps. There could be a default version number so it always works. But I guess the ```ci_with_docker_build.yml ``` script would need to know the version number to pass it in as a build-arg

---

One way to handle it could be to build the project locally using `pip install .` within the docker build workflow, extract the version number with `python -c "import paramak; print(paramak.__version__)"`, and then pass it in as a build-arg as you say. I think the environment variable `SETUPTOOLS_SCM_PRETEND_VERSION` exists to accommodate these sorts of use cases. I just corrected a formatting error I'd introduced in the docker build yml, and I'll see if that's working before playing around with methods that avoid BuildKit.

---

Could we perhaps get the version number from the Git release tag name

---

I got the following to work locally:

```suggestion
ARG paramak_version=develop
RUN SETUPTOOLS_SCM_PRETEND_VERSION_FOR_PARAMAK=${paramak_version} pip install .
```

And then in `docker_publish.yml`, probably something like:

```yml
          build-args: |
            cq_version=2.1
            paramak_version=${{ github.ref_name }}
```

that will set the version to the exact tag which would match the PyPI release (possibly with a leading `v`, depending on how you name your tags). Building locally without setting the version should then just pickup `develop`.

A more robust method might be to use `python -m build` to create a source distribution, `COPY` and `pip install` that into the image. That would always ensure the version was correct, but then you'd probably want to wrap that up in a script for users.

---

I've made a PR to test this build-argument method on the CI
https://github.com/fusion-energy/paramak/pull/159

I also made another option that copies over the .git folder
https://github.com/fusion-energy/paramak/pull/158

Lets see if they both work